pkg/imagestore/imagestore.go

```diff
@@ -268,17 +272,24 @@ func isoFileName(imageType, openshiftVersion, version, arch string) string {
 	return fmt.Sprintf("rhcos-%s-%s-%s-%s.iso", imageType, openshiftVersion, version, arch)
 }
 
-func buildRootfsURL(scheme, host, arch, version string) string {
+func buildRootfsURL(scheme, host, arch, version string) (string, error) {
+	base, err := url.Parse(host)
+	if err != nil {
+		return "", err
+	}
+	if scheme == "" {
+		scheme = base.Scheme
```

Comment:
Should we have another fallback here if the base url doesn't have a scheme? Would `url.Parse` error in that case?

Reply:
In a quick test I did it looks like if there is no scheme the Parse won't fail, but the host gets put into the path so we probably want to watch for that.

Reply:
So worth checking it explicitly or add a comment about it?

Reply:
I think explicitly.

Reply:
Updated. Looks a bit cumbersome though:/